src/test/java/Helloworld/HelloWorldTest.java

```diff
@@ -0,0 +1,33 @@
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import org.junit.Test;
+import static org.junit.Assert.*;
+public class HelloWorldTest
+{
+	String s="hello  world";
+	HelloWorld h=new HelloWorld();
+	@Test
+	public void test1()
+	{
+		assertEquals(s,h.sayHello(s));
+	}
+	//passing null string
+	@Test(expected=IllegalArgumentException.class)
+	public void test2()
+	{
+		h.sayHello(null);
+	}
+	//passing empty string
+	@Test(expected=IllegalArgumentException.class)
+	public void test3()
+	{
+		h.sayHello("");
+	}
+	//passing White spacess
+	@Test(expected=IllegalArgumentException.class)
+	public void test4()
+	{
+		h.sayHello("             ");
```

Comment:
Excellent. There should be a test case for each condition. Only comment is to name your test methods to describe what is being tested. For example, instead of calling this method `test4()`, we can call `testSayHelloWhenInputParamIsWhiteSpaces()`. Don't be afraid to name the methods verbose. Anyone looking at the code should be able to understand by reading the method name.